File: snippets/java/snippet-06.java
@RestController
@RequestMapping("/api")
public class GleanController {
  
  private final Glean gleanClient;
  
  public GleanController(Glean gleanClient) {
    this.gleanClient = gleanClient;
  }
  
  @PostMapping("/chat")
  public ResponseEntity<String> chat(@RequestBody Map<String, String> request) {
    try {
      String message = request.get("message");
      
      var response = gleanClient.client().chat().create()
          .chatRequest(ChatRequest.builder()
              .messages(List.of(
                  ChatMessage.builder()
                      .fragments(List.of(
                          ChatMessageFragment.builder()
                              .text(message)
                              .build()))
                      .build()))
              .build())
          .call();
          
      String responseText = response.chatResponse()
          .map(chat -> chat.messages().stream()
              .flatMap(msg -> msg.fragments().stream())
              .map(fragment -> fragment.text().orElse(""))
              .collect(Collectors.joining()))
          .orElse("No response");
          
      return ResponseEntity.ok(responseText);
    } catch (Exception e) {
      return ResponseEntity.status(500).body("Error: " + e.getMessage());
    }
  }
}

File: snippets/go/snippet-06.go
import (
  "net/http"

  "github.com/labstack/echo/v4"
  apiclientgo "github.com/gleanwork/api-client-go"
  "github.com/gleanwork/api-client-go/models/components"
)

func chatHandler(s *apiclientgo.Glean) echo.HandlerFunc {
  return func(c echo.Context) error {
    var req ChatRequest
    if err := c.Bind(&req); err != nil {
      return echo.NewHTTPError(http.StatusBadRequest, err.Error())
    }

    res, err := s.Client.Chat.Create(c.Request().Context(), components.ChatRequest{
      Messages: []components.ChatMessage{
        {
          Fragments: []components.ChatMessageFragment{
            {Text: apiclientgo.Pointer(req.Message)},
          },
        },
      },
    }, nil, nil)
    if err != nil {
      return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
    }

    answer := ""
    if res.ChatResponse != nil {
      answer = chatAnswer(res.ChatResponse) // helper defined in the Chat API section
    }
    return c.JSON(http.StatusOK, map[string]string{
      "response": answer,
    })
  }
}
